controlplane/eks/api/v1beta1/awsmanagedcontrolplane_types.go

```diff
@@ -166,6 +167,17 @@ type AWSManagedControlPlaneSpec struct { //nolint: maligned
 	// Amazon VPC CNI addon or if you have specified a secondary CIDR block.
 	// +kubebuilder:default=false
 	DisableVPCCNI bool `json:"disableVPCCNI,omitempty"`
+
+	// VPCCNI defines a list of environment variables to apply to the `aws-node` DaemonSet
+	// +optional
+	VPCCNI VPCCNI `json:"VPCCNI,omitempty"`
+}
+
+// VPCCNI specifies configuration related to the VPC CNI.
+type VPCCNI struct {
+	// Environment defines a list of environment variables to apply to the `aws-node` DaemonSet
+	// +optional
+	Environment []corev1.EnvVar `json:"environment,omitempty"`
```

Comment:
I would prefer `Env` (similar to how env vars are defined in a [container](https://github.com/kubernetes/api/blob/7a897301bd567648ff3b967902baafca9f589d8f/core/v1/types.go#L2354))  or `EnvVars` rather than `Environment` as a field name but up to you.